File: frontend/src/types/referenceData.ts
export interface Category { id: number; name: string; description: string | null }
export interface Technician {
  id: number; first_name: string; last_name: string; phone: string; email: string | null
  category_id: number; category_name: string; escalation_order: number; is_team_leader: boolean
}

File: backend/app/schemas/reference_data.py
from pydantic import BaseModel, ConfigDict


class CategoryResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    name: str
    description: str | None


class TechnicianResponse(BaseModel):
    id: int
    first_name: str
    last_name: str
    phone: str
    email: str | None
    category_id: int
    category_name: str
    escalation_order: int
    is_team_leader: bool
